packages/suite-analytics/src/types/events.ts

```diff
@@ -444,4 +444,10 @@ export type SuiteAnalyticsEvent =
           payload: {
               action: 'start' | 'done' | 'learn-more' | 'close-modal';
           };
+      }
+    | {
+          type: EventType.ViewOnlyPromoButton;
+          payload: {
+              action: 'view-only-enabled' | 'view-only-not-set';
```

Comment:
Not sure about this, maybe it doesnt matter but I would just put boolean here and named id `viewOnlyEnabled: true | false`

Reply:
Is there some specific request for this format?

Reply:
But it's not correct what's happening. You are setting the state to `true` or doing nothing. So it could be something like:
`viewOnlyEnabled: true | undefined`, but it's not easy to read

Reply:
But from the semantic of the promo banner you either 

1. accept what is offered to you
2. or you reject what is offered to you


there is no 3rd state => therefore I would suggest boolean

OFC naming may be hard